iml-manager-cli/src/server.rs

```diff
@@ -218,250 +219,400 @@ async fn wait_till_agent_starts(
     Ok(())
 }
 
-#[allow(clippy::cognitive_complexity)]
-pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError> {
-    match command {
-        ServerCommand::List => {
-            let hosts: ApiList<Host> = wrap_fut("Fetching hosts...", get_hosts()).await?;
+fn list_server(hosts: ApiList<Host>) {
+    tracing::debug!("Hosts: {:?}", hosts);
+
+    let table = generate_table(
+        &["Id", "FQDN", "State", "Nids"],
+        hosts.objects.into_iter().map(|h| {
+            vec![
+                h.id.to_string(),
+                h.fqdn,
+                h.state,
+                h.nids.unwrap_or_default().join(" "),
+            ]
+        }),
+    );
+
+    table.printstd();
+}
+
+fn get_profile(config: &AddHosts, profiles: ApiList<ServerProfile>) -> Option<ServerProfile> {
+    let profile_opt = profiles
+        .objects
+        .into_iter()
+        .filter(|x| x.user_selectable)
+        .find(|x| x.name == config.profile);
```

Comment:
Probably, could be `find(x| x.user_selectable && x.name == config.profile)`, without extra `filter`. Theoretically should be faster.

Reply:
But I suspect iterators in Rust are not that simple :)

Reply:
Does the compiler not make this optimization?

Reply:
Compiler should do that.

Reply:
I don't think this is the compiler, but the overall design of iterators: `.filter() returns a new iterator. not a new collection. I have to get used to this :)